ionoscloud/provider.go

```diff
@@ -273,30 +261,48 @@ func Provider() *schema.Provider {
 }
 
 func providerConfigure(d *schema.ResourceData, terraformVersion string) (interface{}, diag.Diagnostics) {
-
-	var clientOpts ClientOptions
-	username, usernameOk := d.GetOk("username")
-	password, passwordOk := d.GetOk("password")
-	token, tokenOk := d.GetOk("token")
+	usernameItf, usernameOk := d.GetOk("username")
+	passwordItf, passwordOk := d.GetOk("password")
+	tokenItf, tokenOk := d.GetOk("token")
+	username := ""
+	if usernameOk {
+		username = usernameItf.(string)
+	}
+	password := ""
+	if passwordOk {
+		password = passwordItf.(string)
+	}
+	token := ""
+	if tokenOk {
+		token = tokenItf.(string)
+	}
 	// for some reason, ENVDEFAULTFUNC does not work for this(boolean?) field
 	if insecure := os.Getenv("IONOS_ALLOW_INSECURE"); insecure != "" {
 		_ = d.Set("insecure", true)
 	}
 	insecure, insecureSet := d.GetOk("insecure")
+	insecureBool := false
 
+	fileConfig, readFileErr := fileconfiguration.NewFromEnv()
 	if !tokenOk {
-		if !usernameOk {
-			diags := diag.FromErr(fmt.Errorf("neither IonosCloud token, nor IonosCloud username has been provided"))
-			return nil, diags
+		if !usernameOk || !passwordOk {
+			if readFileErr != nil {
+				return nil, diag.Errorf("missing credentials, either token or username and password must be set, %s", readFileErr.Error())
+			}
+			profile := fileConfig.GetCurrentProfile()
+			if profile == nil {
+				return nil, diag.Errorf("missing credentials, either token or username and password must be set")
+			}
+			token = profile.Credentials.Token
+			username = profile.Credentials.Username
+			password = profile.Credentials.Password
 		}
-
-		if !passwordOk {
-			diags := diag.FromErr(fmt.Errorf("neither IonosCloud token, nor IonosCloud password has been provided"))
-			return nil, diags
+		if token == "" && (username == "" || password == "") {
+			return nil, diag.Errorf("missing credentials, either token or username and password must be set")
 		}
 	}
 
-	cleanedURL := utils.CleanURL(d.Get("endpoint").(string))
+	endpoint := utils.CleanURL(d.Get("endpoint").(string))
```

Comment:
Just a note that in the future we can use the new sdk func `EnsureURLFormat` (in another PR, not now)